src/effects/effectchainslot.cpp

```diff
@@ -244,6 +244,17 @@ EffectChainPointer EffectChainSlot::getEffectChain() const {
     return m_pEffectChain;
 }
 
+EffectChainPointer EffectChainSlot::getAndEnsureEffectChain(
+        EffectsManager* pEffectsManager) {
+    if (!m_pEffectChain) {
+        auto pEffectChain = EffectChainPointer(
```

Comment:
Same here. The explicit constructor version does not need auto and is more concise.